src/components/can_cooperation/src/validators/struct_validators/module_data_validator.cc

```diff
@@ -83,7 +83,7 @@ ValidationResult ModuleDataValidator::Validate(const Json::Value& json,
                                               outgoing_json[kRadioControlData]);
   } else {
     if (enums_value::kRadio == outgoing_json[kModuleType].asString()) {
```

Comment:
RadioControlData is optional; if it is missed the RPC is still correct. The defect as described(1370) is when controlData IS present and != moduleType. I suggest to change this code into
if (json.isMember(kRadioControlData)) {
  if (enums_value::kRadio != outgoing_json[kModuleType].asString() {
     // ... send error to mobile side
  } else {
    result= RadioControlDataValidator::instnace()->Validate...